Thai charset (#2151)
manticoresoftware/manticoresearch: manual/Creating_a_table/NLP_and_tokenization/Low-level_tokenization.md

charset_table = 0..9, english, _

# you can override character mappings by redefining them, e.g. for case insensitive search with German umlauts you can use:
charset_table = non_cont, U+00E4, U+00C4->U+00E4, U+00F6, U+00D6->U+00F6, U+00FC, U+00DC->U+00FC, U+00DF, U+1E9E->U+00DF
```

<!-- example charset_table -->

> `charset_table` specifies an array that maps letter characters to their case folded versions (or any other characters if you like). The default character set is `non_cont` which includes most languages with [non-continous](https://en.wikipedia.org/wiki/Scriptio_continua) scripts.

Reviewer: ```suggestion
`charset_table` specifies an array that maps letter characters to their case-folded versions (or any other characters if you prefer). The default character set is `non_cont`, which includes most languages with [non-continuous](https://en.wikipedia.org/wiki/Scriptio_continua) scripts.
```

Besides definitions of characters and mappings, there are several built-in aliases that can be used. Current aliases are:
* `english`
* `russian`
* `non_cont`

> * `cont`

Author: Done in https://github.com/manticoresoftware/manticoresearch/pull/2151/commits/8969c3e37785706a48ecfe89f556a4bb008f307a

```
<!-- end -->

> If you do not require support for cont-languages, you can simply exclude the [ngram_len](../../Creating_a_table/NLP_and_tokenization/Low-level_tokenization.md#ngram_len) and [ngram_chars](../../Creating_a_table/NLP_and_tokenization/Low-level_tokenization.md#ngram_chars)

Reviewer: ```suggestion
If you do not require support for continuous-script languages, you can simply exclude the [ngram_len](../../Creating_a_table/NLP_and_tokenization/Low-level_tokenization.md#ngram_len) and [ngram_chars](../../Creating_a_table/NLP_and_tokenization/Low-level_tokenization.md#ngram_chars).
```

<!-- example ngram_len -->
N-gram lengths for N-gram indexing. Optional, default is 0 (disable n-gram indexing). Known values are 0 and 1.

> N-grams provide basic continuous script support for unsegmented texts. The issue with searching in languages using continuous scripts is that there may be no clear separators between the words. In some cases, you may not want to use dictionary-based segmentation [the one available for Chinese](../../Creating_a_table/NLP_and_tokenization/CJK.md). In those cases, n-gram segmentation might work well too.

Reviewer: ```suggestion
N-grams provide basic support for continuous-script languages in unsegmented texts. The issue with searching in languages using continuous scripts is the absence of clear separators between words. In some cases, you may not want to use dictionary-based segmentation, such as [the one available for Chinese](../../Creating_a_table/NLP_and_tokenization/CJK.md). In those instances, n-gram segmentation might also work well.
```
